File: Ahorn/entities/arrowbubble.jl
module IsaGrabBagArrowBubble

using ..Ahorn, Maple

@mapdef Entity "isaBag/arrowBubble" ArrowBubble(x::Integer, y::Integer, direction::String="down")

const directions = String[
	"up",
	"down",
	"left",
	"right"
]

Ahorn.editingOptions(entity::ArrowBubble) = Dict{String, Any}(
	"direction" => directions
)

const placements = Ahorn.PlacementDict(
	"Arrow Bubble (Down, IsaGrabBag)" => Ahorn.EntityPlacement(
		ArrowBubble,
		"rectangle",
		Dict{String, Any}(
			"direction" => "down"
		)
	),
	"Arrow Bubble (Up, IsaGrabBag)" => Ahorn.EntityPlacement(
		ArrowBubble,
		"rectangle",
		Dict{String, Any}(
			"direction" => "up"
		)
	),
	"Arrow Bubble (Left, IsaGrabBag)" => Ahorn.EntityPlacement(
		ArrowBubble,
		"rectangle",
		Dict{String, Any}(
			"direction" => "left"
		)
	),
	"Arrow Bubble (Right, IsaGrabBag)" => Ahorn.EntityPlacement(
		ArrowBubble,
		"rectangle",
		Dict{String, Any}(
			"direction" => "right"
		)
	)
)

function Ahorn.selection(entity::ArrowBubble)
	x, y = Ahorn.position(entity)
	
	return Ahorn.getSpriteRectangle("isafriend/objects/booster/boosterdown00.png", x, y)
	
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::ArrowBubble, room::Maple.Room)
	
	dir = String(get(entity.data, "direction", "down"))
	
	Ahorn.drawSprite(ctx, "isafriend/objects/booster/booster$(dir)00.png", 0, 0)
end

end

File: Loenn/entities/arrow_bubble.lua
local arrowBubble = {}

local directions = {
    Up = "up",
    Down = "down",
    Left = "left",
    Right = "right"
}

arrowBubble.name = "isaBag/arrowBubble"
arrowBubble.depth = -8500
arrowBubble.fieldInformation = {
    direction = {
        options = directions,
        editable = false
    }
}

arrowBubble.placements = {}

for _, dir in pairs(directions) do
    table.insert(arrowBubble.placements, {
        name = dir,
        data = {
            direction = dir
        }
    })
end

function arrowBubble.texture(room, entity)
    return string.format("isafriend/objects/booster/booster%s00", entity.direction)
end

return arrowBubble